TotalCrossSDK/src/main/java/tc/tools/AnonymousUserData.java

```diff
@@ -0,0 +1,144 @@
+     * interrupt the usage of the SDK or pollute the standard output.
+     */
+    public static JSONObject loadConfiguration() {
+        JSONObject config = null;
+        final File configDir = new File(AppDirsFactory.getInstance().getUserConfigDir("TotalCross", null, null)
+                .replace("Application Support", "Preferences")); // this replace only works on macos
+        configDir.mkdirs();
+        configFile = new File(configDir, "config.json");
+        try (FileInputStream fis = new FileInputStream(configFile)) {
+            config = readJsonObject(Stream.asStream(fis));
+        } catch (FileNotFoundException e) {
+            HttpStream.Options options = new HttpStream.Options();
+            options.socketFactory = new SSLSocketFactory();
+            try (HttpStream hs = new HttpStream(new URI(GET_UUID), options)) {
+                config = readJsonObject(hs);
+                config.put("userUuid", config.remove("uuid"));
+
+                try (PrintWriter writer = new PrintWriter(configFile)) {
+                    writer.write(config.toString());
+                }
+            } catch (IOException e1) {
+            }
+        } catch (IOException e) {
+        }
+        return config == null ? new JSONObject() : config;
+    }
+
+    private static JSONObject readJsonObject(Stream stream) throws IOException {
+        try (ByteArrayStream bas = new ByteArrayStream(4096)) {
+            bas.readFully(stream, 10, 4096);
+            JSONObject jsonObject = new JSONObject(new String(bas.getBuffer(), 0, bas.available(), "UTF-8"));
+            return jsonObject;
+        }
+    }
+
+    public void launcher(String... args) {
+        if (!GraphicsEnvironment.isHeadless() && config.isNull("userAcceptedToProvideAnonymousData")) {
+            final String[] options = new String[] { "Yes, send anonymous reports", "Don't send" };
+            final ImageIcon icon = new ImageIcon(getClass().getClassLoader().getResource("tc/crossy.png"));
+            int dialogResult = JOptionPane.showOptionDialog(null, POPUP_TEXT, "", JOptionPane.YES_NO_OPTION,
+                    JOptionPane.QUESTION_MESSAGE, icon, options, options[0]);
+            if (dialogResult == JOptionPane.YES_OPTION) {
+                config.put("userAcceptedToProvideAnonymousData", true);
+            } else if (dialogResult == JOptionPane.NO_OPTION) {
+                config.put("userAcceptedToProvideAnonymousData", false);
+            }
+            try (PrintWriter writer = new PrintWriter(configFile)) {
+                writer.write(config.toString());
+            } catch (FileNotFoundException e) {
+            }
+        }
+        doPost(POST_LAUNCHER, args);
+    }
+
+    public void deploy(String... args) {
+        doPost(POST_DEPLOY, args);
+    }
+
+    private void doPost(String url, String... args) {
+        if (config.optBoolean("userAcceptedToProvideAnonymousData", false)) {
```

Comment:
shouldnt be when userAcceptedToProvideAnonymousData is equal true?

Reply:
ok, I got it! It returns it's value if exists, and return false if does not exist.